src/internal/analytics/components/analytics-funnel.tsx

```diff
@@ -247,5 +257,49 @@ export const AnalyticsFunnelSubStep = ({ children }: AnalyticsFunnelSubStepProps
 
   const context = isNested ? inheritedContext : newContext;
 
+  useEffect(() => {
+    const onMouseDown = () => (mousePressed.current = true);
+
+    const onMouseUp = async () => {
+      mousePressed.current = false;
+
+      if (!isFocusedSubStep.current) {
+        return;
+      }
+
+      /*
+        Some mouse events result in an element being focused. However,
+        this happens only _after_ the onMouseUp event. We yield the
+        event loop here, so that `document.activeElement` has the
+        correct new value.      
+      */
+      await new Promise(r => setTimeout(r, 1));
+
+      if (!subStepRef.current || !subStepRef.current.contains(document.activeElement)) {
+        isFocusedSubStep.current = false;
+
+        /*
+         Run this substep's own focus cleanup function if another substep
+         hasn't already done it for us.
+         */
+        focusCleanupFunction.current?.();
+      }
+    };
+    window.addEventListener('mousedown', onMouseDown);
+    window.addEventListener('mouseup', onMouseUp);
+    return () => {
```

Comment:
Not sure how unlikely is that but if any of the effect dependencies change after mouse it pressed and before it is released, the mousePressed state can remain true which can lead to the wrong step being reported. Should the mousePresses state be reset in the effect cleanup?

Reply:
The `mousePressed` state has the same scope/lifecycle as the effect, so if the effect gets cleaned up because the component unmounts then the `mousePressed` state will also disappear. If the effect is cleaned up only because the dependencies changed, then the new run of the effect will immediately attach new event listeners. So I don't think there is a risk that the `mousePressed` state will get stuck.

Additionally, if we would reset the mousePressed state in the cleanup, and the effect is re-run (e.g. because of strict mode in development, or in React's upcoming "Offscreen rendering"), then the state will be incorrect, because we cannot determine the correct value at the start of the effect.

Reply:
alright, sounds good 👍